packages/nhsuk-frontend/src/nhsuk/components/file-upload/file-upload.mjs

```diff
@@ -0,0 +1,492 @@
+import { closestAttributeValue } from '../../common/closest-attribute-value.mjs'
+import { ConfigurableComponent } from '../../configurable-component.mjs'
+import { ElementError } from '../../errors/index.mjs'
+import { I18n } from '../../i18n.mjs'
+
+/**
+ * File upload component
+ *
+ * @preserve
+ * @augments ConfigurableComponent<FileUploadConfig>
+ */
+export class FileUpload extends ConfigurableComponent {
+  /**
+   * @private
+   * @type {HTMLFileInputElement}
+   */
+  $input
+
+  /**
+   * @private
+   */
+  $button
+
+  /**
+   * @private
+   */
+  $status
+
+  /** @private */
+  i18n
+
+  /** @private */
+  id
+
+  /** @private */
+  $announcements
+
+  /**
+   * @private
```

Comment:
We don't set `@private` or `@internal`. Mainly because these things are never actually private unless we adopt the [private element syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_elements)

Reply:
Probably once [Babel 8](https://next.babeljs.io/docs/v8-migration/) is out and we replace `loose: true` with [compiler assumptions](https://next.babeljs.io/docs/assumptions) [`noUninitializedPrivateFieldAccess`](https://next.babeljs.io/docs/assumptions#nouninitializedprivatefieldaccess) and [`privateFieldsAsProperties`](https://next.babeljs.io/docs/assumptions#privatefieldsasproperties)

You can see it was tried in the past https://github.com/alphagov/govuk-frontend/pull/4099/commits/131a0874eb345f8659aef2fc235b4de5c7c334f2 but it did add a reasonable overhead to the compiled code so GOV.UK Frontend just pretends they're private methods